cmd/ncproxy/server.go

```diff
@@ -88,3 +119,59 @@ func (s *server) serve(ctx context.Context, ttrpcListener net.Listener, grpcList
 		serveErr <- trapClosedConnErr(s.grpc.Serve(grpcListener))
 	}()
 }
+
+// reconnectComputeAgents creates new compute agent connections from the database of
+// active compute agent addresses and adds them to the compute agent client cache
+// this MUST be called before the server start serving anything so that we can
+// ensure that the cache is ready when they do.
```

Comment:
This comment or some other location could use additional detail how ncproxy reconnect is intended to work. Documenting the database schema would be helpful too.

Reply:
What additional information do you think would be helpful to detail how the reconnect flow works?

Reply:
Something like:

reconnectComputeAgents
Ncproxy maintains a cache of active compute agents in order reestablish connections if the service is restarted. The cache is persisted in a bolt database with the following schema: 

On restart ncproxy will attempt to create new compute agent connections from the database of active compute agent addresses and add them to its compute agent client cache. Reconnect MUST be called before the server is allowed to start serving anything so that we can ensure that the cache is ready. Reconnections are performed in parallel to improve service startup performance.

There are a few failure modes for reconnect. If a compute agent entry is stale the reconnect will fail in the following way:  xxxxx. Other failure modes are possible but not expected. In those cases we log the failures but allow the service start to proceed. We chose this approach vs just failing service start for the following reasons xxxx.

Reply:
Added!